include/usb/class/usb_hid.h

```diff
@@ -54,6 +54,9 @@ struct usb_hid_descriptor {
 typedef int (*hid_cb_t)(struct usb_setup_packet *setup, s32_t *len,
 			u8_t **data);
 typedef void (*hid_int_ready_callback)(void);
+#ifdef CONFIG_ENABLE_HID_INT_OUT_EP
+typedef void (*hid_out_ready_callback)(void *data, u32_t len);
```

Comment:
Please make `hid_out_ready_callback` a notification similar to `hid_int_ready_callback` and expect the application to do the read.

Reply:
I have tried that implementation. I was not able receive the correct data sent from host. so I had put read in callback and added application call back.

Reply:
Could it be due to the `hid_ep_data[]` contents being corrupted per my other comment?

Reply:
no. I tried with hid_ep_data[] after changing as per your comment. still that issue is present.